### PR TITLE
Rust 1.83 as well update outdated dependencies

### PR DESCRIPTION
**Pull request by jcowgar**
* Updated to Rust 1.83
* Used cargo outdated to identify and upgrade all outdated dependencies
* Fixed issues that newer versions of clippy reported

This has solved some compilation warnings for packages such as wasmtime. It also keeps us on a recent version of Rust without getting too far behind making an upgrade more difficult.

### CODE REVIEW

File: src/buffer.rs [hunk 1]
```diff
@@ -23,7 +23,7 @@ use shared::{
 };
 use std::{collections::HashSet, ops::Range};
 use tree_sitter::{Node, Parser, Tree};
-use tree_sitter_traversal::{traverse, Order};
+use tree_sitter_traversal2::{traverse, Order};
```
Comment (wongjiahau) [inline src/buffer.rs:26]: Why is this necessary?

File: src/lsp/process.rs [hunk 2]
```diff
@@ -295,12 +295,7 @@ impl LspServerProcess {
             None,
             InitializeParams {
                 process_id: None,
-                root_uri: Some(Url::parse(&format!(
-                    "file://{}",
-                    self.current_working_directory.display_absolute()
-                ))?),
```
Comment (wongjiahau) [inline src/lsp/process.rs:301]: Why is this removed?

File: src/lsp/process.rs [hunk 3]
```diff
@@ -399,7 +394,13 @@ impl LspServerProcess {
...
                     ..ClientCapabilities::default()
                 },
-                workspace_folders: None,
+                workspace_folders: Some(vec![WorkspaceFolder {
+                    uri: Url::parse(&format!(
+                        "file://{}",
+                        self.current_working_directory.display_absolute()
+                    ))?,
+                    name: "root".to_string(),
+                }]),
```
Comment (wongjiahau) [inline src/lsp/process.rs:403]: Why is this added?

File: src/lsp/process.rs (hunk 2 above)
Comment (jcowgar) [reply src/lsp/process.rs:301]: `root_uri` was deprecated in `lsp_types` in favor of `workspace_folders`. With `root_uri` still in the code a warning was omitted about the deprecation.

See: https://docs.rs/lsp-types/0.97.0/lsp_types/struct.InitializeParams.html#structfield.root_uri

**Comment by wongjiahau** [comment]
Is this ready to be merged?

**Comment by jcowgar** [comment]
Yes. I have been using it since I created it, no problems so far and tests pass. Didn't want to merge this w/o your approval/input.
